internal/core/operations/switch_active_version/switch_active_version_executor_test.go

```diff
@@ -0,0 +1,382 @@
+
+			currentGameRoomInstance := game_room.Instance{
+				ID:          roomId,
+				SchedulerID: definition.NewActiveScheduler.Name,
+			}
+
+			newGameRoomInstance := game_room.Instance{
+				ID:          fmt.Sprintf("new-%s", roomId),
+				SchedulerID: definition.NewActiveScheduler.Name,
+			}
+
+			newGameRoom := game_room.GameRoom{
+				ID:          newGameRoomInstance.ID,
+				SchedulerID: definition.NewActiveScheduler.Name,
+				Status:      game_room.GameStatusPending,
+				Version:     newScheduler.Spec.Version,
+			}
+
+			mocks.portAllocator.EXPECT().Allocate(gomock.Any(), 1).Return([]int32{5000}, nil)
+			mocks.runtime.EXPECT().CreateGameRoomInstance(context.Background(), definition.NewActiveScheduler.Name, versionEq(newScheduler.Spec.Version)).Return(&newGameRoomInstance, nil)
+
+			gameRoomReady := newGameRoom
+			gameRoomReady.Status = game_room.GameStatusReady
+			gameRoomTerminating := currentGameRoom
+			gameRoomTerminating.Status = game_room.GameStatusTerminating
+			mocks.roomStorage.EXPECT().CreateRoom(gomock.Any(), gomock.All(idEq(newGameRoom.ID), versionEq(newScheduler.Spec.Version))).Return(nil)
+			mocks.roomStorage.EXPECT().GetRoom(gomock.Any(), gameRoomReady.SchedulerID, gameRoomReady.ID).Return(&gameRoomReady, nil)
+			roomStorageStatusWatcher := room_storage_mock.NewMockRoomStorageStatusWatcher(mockCtrl)
+			mocks.roomStorage.EXPECT().WatchRoomStatus(gomock.Any(), gomock.All(idEq(newGameRoom.ID), versionEq(newScheduler.Spec.Version))).Return(roomStorageStatusWatcher, nil)
+
+			mocks.instanceStorage.EXPECT().GetInstance(gomock.Any(), definition.NewActiveScheduler.Name, roomId).Return(&currentGameRoomInstance, nil)
+			mocks.roomStorage.EXPECT().WatchRoomStatus(gomock.Any(), gomock.All(idEq(currentGameRoom.ID), versionEq(currentVersion))).Return(roomStorageStatusWatcher, nil)
+			mocks.roomStorage.EXPECT().GetRoom(gomock.Any(), gameRoomTerminating.SchedulerID, gameRoomTerminating.ID).Return(&gameRoomTerminating, nil)
+			mocks.runtime.EXPECT().DeleteGameRoomInstance(gomock.Any(), &currentGameRoomInstance).Return(nil)
+
+			roomStorageStatusWatcher.EXPECT().Stop().Times(2)
+		}
+
+		executor := switch_active_version.NewExecutor(mocks.roomManager, mocks.schedulerManager)
+		err = executor.Execute(context.Background(), &operation.Operation{}, definition)
+		require.Error(t, err)
+	})
+}
+
+func TestSwitchActiveVersionOperation_OnError(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	mocks := newMockRoomAndSchedulerManager(mockCtrl)
+
+	err := validations.RegisterValidations()
+	if err != nil {
+		t.Errorf("unexpected error %d'", err)
+	}
+
+	newScheduler := newValidScheduler()
+	definition := &switch_active_version.SwitchActiveVersionDefinition{
+		NewActiveScheduler: newScheduler,
+	}
+
+	t.Run("should succeed - Execute on error if operation finishes (no created rooms)", func(t *testing.T) {
```

Comment:
@caiordjesus I think the major feature of the OnError logic is to remove rooms that were created, right? what about adding a test case exploring this scenario here?